test/prototype_transforms_dispatcher_infos.py

```diff
@@ -0,0 +1,179 @@
+import dataclasses
+from typing import Callable, Dict, Type
+
+import pytest
+import torch
+import torchvision.prototype.transforms.functional as F
+from prototype_common_utils import ArgsKwargs
+from prototype_transforms_kernel_infos import KERNEL_INFOS
+from test_prototype_transforms_functional import FUNCTIONAL_INFOS
+from torchvision.prototype import features
+
+__all__ = ["DispatcherInfo", "DISPATCHER_INFOS"]
+
+KERNEL_SAMPLE_INPUTS_FN_MAP = {info.kernel: info.sample_inputs_fn for info in KERNEL_INFOS}
+
+
+# Helper class to use the infos from the old framework for now tests
+class PreloadedArgsKwargs(ArgsKwargs):
+    def load(self, device="cpu"):
+        args = tuple(arg.to(device) if isinstance(arg, torch.Tensor) else arg for arg in self.args)
+        kwargs = {
+            keyword: arg.to(device) if isinstance(arg, torch.Tensor) else arg for keyword, arg in self.kwargs.items()
+        }
+        return args, kwargs
+
+
+KERNEL_SAMPLE_INPUTS_FN_MAP.update(
+    {
+        info.functional: lambda: (
+            PreloadedArgsKwargs(*args_kwargs.args, **args_kwargs.kwargs) for args_kwargs in info.sample_inputs()
+        )
+        for info in FUNCTIONAL_INFOS
+    }
+)
+
+
+@dataclasses.dataclass
+class DispatcherInfo:
+    dispatcher: Callable
+    kernels: Dict[Type, Callable]
+
+    def sample_inputs(self, *types):
+        for type in types or self.kernels.keys():
+            if type not in self.kernels:
+                raise pytest.UsageError(f"There is no kernel registered for type {type.__name__}")
+
+            yield from KERNEL_SAMPLE_INPUTS_FN_MAP[self.kernels[type]]()
+
+
+DISPATCHER_INFOS = [
+    DispatcherInfo(
+        F.horizontal_flip,
+        kernels={
+            features.Image: F.horizontal_flip_image_tensor,
+            features.BoundingBox: F.horizontal_flip_bounding_box,
+            features.Mask: F.horizontal_flip_mask,
+        },
```

Comment:
Given that we are using a strict naming scheme, we could later also try to fill this dictionary automatically. For now let's be explicit.